Docs (#38)
smaht-dac/smaht-portal: docs/source/excel_submission.rst

> .. role:: raw-html-m2r(raw)

Reviewer: Looks like a lot of information in this file especially that needs to be redone for smaht, particularly from lines 145 downward which refers to 4DN-specific tools/ideas. If we're not going to show this page, then might be fine to leave in with a TODO about updating these.

   :format: html

================
Excel Submission
================

Overview
--------

> * Metadata and data can be submitted to our platform using Microsoft Excel WorkBooks that describe related items in separate sheets.

Reviewer: ```suggestion
* Metadata can be submitted to our platform using Microsoft Excel WorkBooks that describe related items in separate sheets.
```

* This section provides detailed information on how to use the WorkBooks.
* A link to a detailed example of a filled out spreadsheet is forthcoming.
* Based on the type of experiment(s) for which you plan to submit data, the data wranglers can provide you with an Excel WorkBook containing several WorkSheets.
* Each sheet corresponds to an Item type in our metadata database.
* The workbook provided should contain all the sheets that you may need for your submission.
* Each sheet should also contain all the data fields that can be submitted for that Item type.

> * Depending on if you have submitted data before or if you are using shared reagents that have been submitted by other labs, you may not need to provide information on every sheet or in every field.

Reviewer: ```suggestion
* Depending on if you have submitted data before or if you are using shared resources that have been submitted by other groups, you may not need to provide information on every sheet or in every field.
```

**Organization of the Workbook**

* Generally, it makes sense to begin with the left most sheet in the workbook as the sheets in a workbook are ordered so that Items that have fields that take a reference to another Item as their value appear ‘after’ i.e. to the right of that Item’s sheet in the workbook.
* A sheet for an Item starts with a row of field names.
* *Absolutely required fields are marked with a leading asterisk (eg. \*\ experiment_type).* - failure to supply a value in these fields will cause an error
* The second row of the sheet indicates the type of the information expected for the fields.
* The third row includes a description of each of the fields.
* In some cases the values that you can submit for a particular field are constrained to a specific set of terms and when this is the case the possible values are shown in the fourth row.
* Any row that starts with “#” in the first column will be ignored, so you can add non-data rows for your own use.

> * However, **PLEASE NOTE THAT THE FIRST 2 ROWS OF A SHEET SHOULD NOT BE MODIFIED.**

Reviewer: I'm not sure how valid all this information is for us vs. 4DN here. Consider excluding until such things are settled.

  * In many cases the values of these fields need to be in database_name:ID format. eg. an SRA experiment identifier ‘SRA:SRX1234567’ (see also `Basic fields example <#basic-field>`_ above).
  * In a few cases where the field takes only identifiers for one specific databases the ID alone can be entered - eg. *'targeted_genes’* field of the Target Item enter gene symbols eg. PARK2, DLG1.

Linked item fields
^^^^^^^^^^^^^^^^^^

* Some fields in a Sheet for an Item may contain references to another Item.
* The referenced Item may be of the same or different type.
* The *'files'* field is also an example of a list field that can take multiple values.
* You can reference an item in the excel workbooks using one of four possible ways:

  #. submitted_id
  #. accession
  #. item-type-specific identifier
  #. UUID

We recommend using submitted_id for simplicity. More information about these four identifiers is provided in `Using aliases </help/submitter-guide/getting-started#using-identifiers>`_.

Field(s) with subobjects
^^^^^^^^^^^^^^^^^^^^^^^^

TODO: handled differently in the new submission tool and should be documented

Referencing existing items
^^^^^^^^^^^^^^^^^^^^^^^^^^

> * Ways that you can reference items that already exist in the 4DN database in your spreadsheet submission is described `here </help/submitter-guide/getting-started#referencing-existing-objects>`_.
>   :raw-html-m2r:`<span id="supp_files"></span>`
> * In some cases information for existing items will be present in the Excel Work Sheets provided for your submission.
> * You can also check the existing items from *collection* pages that list all of them.
> * The links for item lists can be constructed by ``https://data.4dnucleome.org/ + plural-object-name`` (e.g. https://data.smaht.org/analyte/ ) and the identifiers that can be used for collections are referenced in `this table <schema_info.md>`_.

Reviewer: ```suggestion
* Ways that you can reference items that already exist in the SMaHT database in your spreadsheet submission is described `here </help/submitter-guide/getting-started#referencing-existing-objects>`_.
  :raw-html-m2r:`<span id="supp_files"></span>`
* In some cases information for existing items will be present in the Excel Work Sheets provided for your submission.
* You can also check the existing items from *collection* pages that list all of them.
* The links for item lists can be constructed by ``https://data.smaht.org/ + plural-object-name`` (e.g. https://data.smaht.org/analyte/ ) and the identifiers that can be used for collections are referenced in `this table <schema_info.md>`_.
```
